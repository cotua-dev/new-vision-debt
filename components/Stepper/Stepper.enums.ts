export enum UserPurpose {
    ReduceYourMonthlyPayments = 80,
    ConsolidateYourPayments = 81,
    ReduceYourInterestRates = 82,
    AllOfTheAbove = 83,
};

export enum DebtType {
    CreditCards = 84,
    PersonalLoans = 85,
    MedicalBills = 86,
    AllOfTheAbove = 87,
};

export enum BehindPaymentsType {
    YesMore60 = 88,
    YesMore30 = 89,
    No = 90,
};

export enum FallBehindReason {
    LossOfIncome = 91,
    IncreasedCostOfLiving = 92,
    UnexpectedExpenses = 93,
    MedicalHardship = 94,
};

export enum Questions {
    UserPurpose = 'What are you looking to do?',
    DebtType = 'What kind of debts do you have?',
    BehindPaymentsType = 'Are you behind on your payments?',
    FallBehindReason = 'What made you fall behind?',
    MonthlyIncomeAmount = 'What is your monthly income?',
    UnsecuredDebtAmount = 'What is the total amount of your unsecured debt?',
    ZipCode = 'What city are you in?',
    Name = 'What is your name?',
    Email = 'What is your email?',
    Phone = 'Please provide your phone number for verification',
    Code = 'Please provide the code sent to your phone number',
};
