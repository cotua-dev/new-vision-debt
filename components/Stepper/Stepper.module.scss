@use 'styles/variables';
@use 'styles/mixins';

section.debt-advisors-stepper {
    @include mixins.base-section;
    @include mixins.flexify(center, center, column);
    height: 550px;
    padding: 10px;

    @include mixins.mobile {
        height: 100%;
        padding: 0 10px 50px;
    }

    div.steps-wrapper {
        div.step {
            text-align: center;
            margin: 50px 10px;

            @include mixins.custom(767px) {
                margin: 20px 10px;
            }

            &.hidden {
                display: none;
            }

            strong.question {
                @include mixins.main-heading;

                @include mixins.mobile {
                    font-size: 24px;
                }
            }
        }
    }

    small.error {
        font-size: 14px;
        font-weight: 700;
        color: variables.$secondary-color;
    }

    div.stepper-controls-wrapper {
        margin-top: 20px;
        display: flex;
        flex-direction: row;

        @include mixins.mobile {
            width: 100%;
        }

        button.previous-button,
        button.next-button {
            cursor: pointer;
            border: 0;
            border-radius: 10px;
            padding: 10px 0;
            color: #FFFFFF;
            font-family: 'Montserrat', Arial, Helvetica, sans-serif;
            font-size: 18px;
            font-weight: 300;
            @include mixins.flexify(center, center, row);

            &:disabled {
                cursor: not-allowed;
            }

            .caret {
                height: 20px;
                width: 20px;
            }
        }

        button.previous-button {
            width: 80px;
            background-color: variables.$secondary-color;

            &:disabled {
                background-color: lighten(variables.$secondary-color, 20%);
            }
        }

        button.next-button {
            width: 300px;
            background-color: variables.$primary-color;
            margin-left: 20px;

            &:disabled {
                background-color: lighten(variables.$primary-color, 20%);
            }

            @include mixins.mobile {
                width: 100%;
            }

            .caret {
                margin-left: 10px;
            }
        }
    }
}
