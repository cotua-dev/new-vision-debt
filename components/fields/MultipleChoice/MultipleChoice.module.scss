@use 'styles/variables';
@use 'styles/mixins';

div.multiple-choice-wrapper {
    width: 100%;
    @include mixins.flexify(center, center, row);
    flex-wrap: wrap;
    margin: 20px 0;

    @include mixins.mobile {
        flex-direction: column;
    }

    button.choice-button {
        background-color: #FFFFFF;
        width: 250px;
        border: 0;
        border-radius: 10px;
        padding: 30px 0;
        margin: 10px;
        cursor: pointer;
        box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
        transition: all 250ms;

        &:hover {
            box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
            scale: 1.05;
        }

        &.active {
            background-color: variables.$primary-color;
            box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
            scale: 1.05;
            
            strong.choice-text {
                color: #FFFFFF;
            }
        }

        @include mixins.mobile {
            width: 100%;
            margin: 10px 0;
            padding: 20px;
        }

        strong.choice-text {
            color: variables.$primary-color;
            font-size: 14px;
            font-weight: 700;
            text-align: center;
        }
    }
}
