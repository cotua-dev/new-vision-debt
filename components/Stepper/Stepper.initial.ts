import { StepperModel, Step } from './Stepper.interfaces';
import { Questions } from './Stepper.enums';

export const initialStepperModel: StepperModel = {
    userPurpose: null,
    debtType: null,
    behindPaymentsType: null,
    fallBehindReason: null,
    monthlyIncomeAmount: '',
    unsecuredDebtAmount: '',
    zipCode: '',
    firstName: '',
    lastName: '',
    email: '',
    phone: '',
    code: '',
    isMailer: false,
    sourceSite: '',
};

export const startingSteps: Step[] = [
    {
        question: Questions.UserPurpose,
        validity: false,
        property: ['userPurpose'],
        stepType: 'multiple-choice',
        choices: [
            { label: 'Reduce your monthly payments', value: 0 },
            { label: 'Consolidate your payments', value: 1 },
            { label: 'Reduce your interest rates', value: 2 },
            { label: 'All of the above', value: 3 },
        ],
        value: {},
    },
    {
        question: Questions.DebtType,
        validity: false,
        property: ['debtType'],
        stepType: 'multiple-choice',
        choices: [
            { label: 'Credit cards', value: 0 },
            { label: 'Personal loans', value: 1 },
            { label: 'Medical bills', value: 2 },
            { label: 'All of the above', value: 3 },
        ],
        value: {},
    },
    {
        question: Questions.BehindPaymentsType,
        validity: false,
        property: ['behindPaymentsType'],
        stepType: 'multiple-choice',
        choices: [
            { label: 'Yes - more than 60 days', value: 0 },
            { label: 'Yes - more than 30 days', value: 1 },
            { label: 'No', value: 2 },
        ],
        value: {},
    },
];

export const fallBehindReasonStep: Step[] = [
    {
        question: Questions.FallBehindReason,
        validity: false,
        property: ['fallBehindReason'],
        stepType: 'multiple-choice',
        choices: [
            { label: 'Loss of income', value: 0 },
            { label: 'Increase in cost of living', value: 1 },
            { label: 'Unexpected expenses', value: 2 },
            { label: 'Medical hardship', value: 3 },
        ],
        value: {},
    },
];

export const userSteps: Step[] = [
    {
        question: Questions.ZipCode,
        validity: false,
        property: ['zipCode'],
        stepType: 'location',
        value: {},
    },
    {
        question: Questions.Name,
        validity: false,
        property: ['firstName', 'lastName'],
        stepType: 'name',
        value: {},
    },
    {
        question: Questions.Email,
        validity: false,
        property: ['email'],
        stepType: 'email',
        value: {},
    },
    {
        question: Questions.Phone,
        validity: false,
        property: ['phone'],
        stepType: 'phone',
        value: {},
    },
    {
        question: Questions.Code,
        validity: false,
        property: ['code'],
        stepType: 'verify',
        value: {},
    },
];

export const endingSteps: Step[] = [
    {
        question: Questions.MonthlyIncomeAmount,
        validity: false,
        property: ['monthlyIncomeAmount'],
        stepType: 'currency',
        value: {},
    },
    {
        question: Questions.UnsecuredDebtAmount,
        validity: false,
        property: ['unsecuredDebtAmount'],
        stepType: 'currency',
        value: {},
    },
];

export const initialSteps: Step[] = [
    ...startingSteps,
    ...endingSteps,
    ...userSteps,
];

export const initialNoUserSteps: Step[] = [
    ...startingSteps,
    ...endingSteps,
];
