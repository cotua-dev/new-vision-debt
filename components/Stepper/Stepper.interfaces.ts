import { MultipleChoiceValues, StepperType, StepTypes } from './Stepper.types';
import { UserPurpose, DebtType, BehindPaymentsType, FallBehindReason, Questions } from './Stepper.enums';
import { MailerCheckResponseBody } from '../MailerCheck/MailerCheck.interfaces';

export interface MultipleChoice {
    label: string;
    value: MultipleChoiceValues;
};

export interface MailerModel {
    referenceId: string;
    middleInitial: string;
    suffix: string;
    address: string;
    city: string;
    stateAbbreviation: string;
    crrt: string;
    barcode: string;
    county: string;
    estimatedDebt: number;
    settled: number;
    newPayment: number;
    tier: string;
    segment: string;
    drop: string;
    tollFreeNumber: string;
    spanishTollFreeNumber: string;
    noticeDate: string;
    url: string;
    stateFull: string;
};

export interface StepperModel extends Partial<MailerModel> {
    userPurpose: MultipleChoiceValues | null;
    debtType: MultipleChoiceValues | null;
    behindPaymentsType: MultipleChoiceValues | null;
    fallBehindReason: MultipleChoiceValues | null;
    monthlyIncomeAmount: string;
    unsecuredDebtAmount: string;
    zipCode: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    code: string;
    isMailer: boolean;
};

export interface ParsedStepperModel extends Partial<MailerModel> {
    userPurpose: UserPurpose;
    debtType: DebtType;
    behindPaymentsType: BehindPaymentsType;
    fallBehindReason: FallBehindReason;
    monthlyIncomeAmount: number;
    unsecuredDebtAmount: number;
    zipCode: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    site: string;
    isMailer: boolean;
};

export interface Step {
    question: Questions;
    validity: boolean;
    property: (keyof StepperModel)[];
    stepType: StepTypes;
    choices?: MultipleChoice[];
    value: { [StepperModelKey in keyof StepperModel]?: null | string | number; };
};

export interface StepperProps {
    'stepper-type': StepperType,
    'user-info'?: MailerCheckResponseBody,
};
