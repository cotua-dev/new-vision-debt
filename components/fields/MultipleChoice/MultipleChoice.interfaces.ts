import { Dispatch, SetStateAction } from 'react';
import { MultipleChoice, Step } from '../../Stepper/Stepper.interfaces';
import { StepperType } from '../../Stepper/Stepper.types';
import { SharedFieldProps } from '../shared.interfaces';

export interface MultipleChoiceFieldProps extends SharedFieldProps {
    choices: MultipleChoice[];
    nextStep: Function;
    setSteps: Dispatch<SetStateAction<Step[]>>;
    stepperType: StepperType,
};
