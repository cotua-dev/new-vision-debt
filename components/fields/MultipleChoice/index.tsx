import styles from './MultipleChoice.module.scss';
import { MultipleChoiceFieldProps } from './MultipleChoice.interfaces';
import { MultipleChoice, Step, StepperModel } from '../../Stepper/Stepper.interfaces';
import { MultipleChoiceValues } from '../../Stepper/Stepper.types';
import { Questions } from '../../Stepper/Stepper.enums';
import { endingSteps, fallBehindReasonStep, startingSteps, userSteps } from '../../Stepper/Stepper.initial';

/**
 * Component for handling multiple choice fields
 * @param props Object containing the component properties
 * @returns JSX element markup
 */
export function MultipleChoiceField(props: MultipleChoiceFieldProps): JSX.Element {
    // Grab what we need from props
    const { model, field, setModel, nextStep, choices, steps, currentStep, setSteps } = props;

    // We know there will only ever be one field for multiple choice
    const fieldName: keyof StepperModel = field[0];

    /**
     * Set the value in the model
     * @param value Number between 0 and 3 inclusive
     */
    function setValue(value: MultipleChoiceValues): void {
        // Build the new model using the first passed field
        const newModel: StepperModel = { ...model, [fieldName]: value };

        // Set the new model
        setModel(newModel);

        // Make the current step valid
        steps[currentStep].validity = true;

        // Get the current step object
        const currentStepObject: Step = steps[currentStep];

        // Check to see if we are at the behind payments type question
        if (currentStepObject.question === Questions.BehindPaymentsType) {
            let stepsToAdd: Step[] = [];

            // If 'No' is not selected, then add the fall behind reason question
            if (value !== 2) {
                stepsToAdd = [
                    ...startingSteps,
                    ...fallBehindReasonStep,
                    ...endingSteps,
                ];
            } else {
                // Otherwise, keep the fall behind reason question out
                stepsToAdd = [
                    ...startingSteps,
                    ...endingSteps,
                ];

                // Reset the fall behind reason value and set the behind payments type value
                setModel({
                    ...model,
                    behindPaymentsType: value,
                    fallBehindReason: null,
                });
            }

            if (props.stepperType === 'full') {
                stepsToAdd = [...stepsToAdd, ...userSteps];
            }

            setSteps(stepsToAdd);
        }

        // Update local storage
        // window.localStorage.setItem('model', JSON.stringify(newModel));

        // Go to the next step
        nextStep();
    }

    return (
        <div className={styles['multiple-choice-wrapper']}>
            {choices.map((choice: MultipleChoice) => (
                <button
                    className={`${styles['choice-button']} ${model[fieldName] === choice.value ? styles['active'] : ''}`}
                    key={choice.value}
                    type="button"
                    onClick={() => setValue(choice.value)}
                >
                    <strong className={styles['choice-text']}>
                        {choice.label}
                    </strong>
                </button>
            ))}
        </div>
    );
}
