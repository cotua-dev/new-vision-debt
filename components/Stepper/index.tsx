import { useState } from 'react';
import { useRouter } from 'next/router';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleRight, faAngleLeft, faCircleNotch } from '@fortawesome/free-solid-svg-icons';
import styles from './Stepper.module.scss';
import { ParsedStepperModel, Step, StepperModel, StepperProps } from './Stepper.interfaces';
import { initialNoUserSteps, initialStepperModel, initialSteps } from './Stepper.initial';
import { addBitrixContactDeal, parseCurrencyValue, parseModel, sendSMS, verifySMSCode } from './Stepper.utility';
import { Questions } from './Stepper.enums';
import { MultipleChoiceField } from '../fields/MultipleChoice';
import { Currency } from '../fields/Currency';
import { Location } from '../fields/Location';
import { Name } from '../fields/Name';
import { Email } from '../fields/Email';
import { Phone } from '../fields/Phone';
import { Verify } from '../fields/Verify';

export function Stepper(props: StepperProps): JSX.Element {
    const router = useRouter();
    const [model, setModel] = useState(initialStepperModel);
    const [currentStep, setCurrentStep] = useState(0);
    const [steps, setSteps] = useState(props['stepper-type'] == 'full' ? initialSteps : initialNoUserSteps);
    const [disableNextButton, setDisableNextButton] = useState(true);
    const [disablePreviousButton, setDisablePreviousButton] = useState(true);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState('');
    const [disableVerifyField, setDisableVerifyField] = useState(true);

    // Fields
    const [zipCode, setZipCode] = useState(props['user-info'] === undefined ? '' : props['user-info'].zip);
    const [firstName, setFirstName] = useState(props['user-info'] === undefined ? '' : props['user-info'].firstName);
    const [lastName, setLastName] = useState(props['user-info'] === undefined ? '' : props['user-info'].lastName);
    const [email, setEmail] = useState('');
    const [phone, setPhone] = useState('');
    const [code, setCode] = useState('');
    const [usState, setUSState] = useState(props['user-info'] === undefined ? '' : props['user-info'].stateAbbreviation);

    /**
     * Send a SMS to the provided phone number to verify the user owns the number
     */
    async function kickOffSMS(): Promise<void> {
        // Show a loading spinner, disable verify field, and reset error
        setLoading(true);
        setDisableVerifyField(true);
        setError('');

        // Make sure the phone is not an empty string
        if (phone !== '') {
            const smsResponse: Response | undefined = await sendSMS(phone);

            // Make sure we have a response object and its status is 200
            if (smsResponse !== undefined && smsResponse.status === 200) {
                // Enable verify field
                setDisableVerifyField(false);
            } else {
                // Display error
                setError('Something went wrong. Please try again');
            }
        } else {
            // Display error
            setError('Please provide a valid phone number');
        }

        // Hide the loading spinner
        setLoading(false);
    }

    function sendThankYouPage() {
        switch (router.pathname) {
            case '/ohio':
                window.location.href = `${window.location.origin}/ohio/thank-you`;
                break;
            case '/new-jersey':
                window.location.href = `${window.location.origin}/new-jersey/thank-you`;
                break;
            case '/colorado':
                window.location.href = `${window.location.origin}/colorado/thank-you`;
                break;
            case '/credit-card':
                window.location.href = `${window.location.origin}/credit-card/thank-you`;
                break;
            case '/debt':
                window.location.href = `${window.location.origin}/debt/thank-you`;
                break;
            case '/medical':
                window.location.href = `${window.location.origin}/medical/thank-you`;
                break;
            case '/personal':
                window.location.href = `${window.location.origin}/personal/thank-you`;
                break;
            case '/student-loan':
                window.location.href = `${window.location.origin}/student-loan/thank-you`;
                break;
            default:
                window.location.href = `${window.location.origin}/thank-you`;
                break;
        }
    }

    /**
     * Verify the user provided code with their phone number
     */
    async function submission(data: ParsedStepperModel): Promise<void> {
        // Show a loading spinner and reset error
        setLoading(true);
        setError('');

        if (props['stepper-type'] === 'short') {
            const bitrixResponse: Response | undefined = await addBitrixContactDeal(data);

            if (bitrixResponse !== undefined && bitrixResponse.status === 201) {
                sendThankYouPage();
            } else {
                setError('Something went wrong. Please try again');
            }

            // Stop the function here
            setLoading(false);
            return;
        }

        // Make sure we have a `phone` and `code`
        if (phone !== '' && code !== '' && props['stepper-type'] === 'full') {
            const verifyResponse: Response | undefined = await verifySMSCode(phone, code);

            // Make sure we have a response object and its status is 200
            if (verifyResponse !== undefined && verifyResponse.status === 200) {
                const bitrixResponse: Response | undefined = await addBitrixContactDeal(data);

                // Make sure we have a response object and its status is 201
                if (bitrixResponse !== undefined && bitrixResponse.status === 201) {
                    // Send to thank you page with browser refresh (this way state is completely wiped in one go)
                    sendThankYouPage();
                } else {
                    // Display error
                    setError('Something went wrong. Please try again');
                }
            } else {
                // Display error
                setError('Something went wrong. Please try again');
            }
        } else {
            // Display error
            setError('Please provide a valid phone number and code for verification');
        }

        // Hide the loading spinner
        setLoading(false);
    }

    /**
     * @todo Needs work
     */
    function previousStep() {
        if (currentStep - 1 < 0) {
            setDisablePreviousButton(true);
            setCurrentStep(0);
        } else {
            setDisablePreviousButton(false);
            setCurrentStep(currentStep - 1);
        }

        setDisableNextButton(false);
    }

    /**
     * @todo Needs work
     */
    function nextStep() {
        // Grab needed properties from the current step
        const { validity, question } = steps[currentStep];

        // Check if we passed the unsecured debt field
        if (question === Questions.UnsecuredDebtAmount) {
            // Parse the unsecured debt amount into a number
            const unsecuredDebtAmountNumber: number = parseCurrencyValue(model.unsecuredDebtAmount);

            // Check if the debt is less than $5,000
            if (unsecuredDebtAmountNumber < 5000) {
                // If so, send to disqualify page with browser refresh (this way state is completely wiped in one go)
                window.location.href = `${window.location.origin}/dq`;
            }
        }

        // Check if we passed the zip code field
        if (question === Questions.ZipCode) {
            // Check to see if the city is in North Dakota or South Carolina
            if (usState === 'North Dakota' || usState === 'South Carolina') {
                // If so, send to disqualify page with browser refresh (this way state is completely wiped in one go)
                window.location.href = `${window.location.origin}/dq`;
            }
        }

        // Check if we passed the phone number field
        if (question === Questions.Phone) {
            // We should have a phone number at this point which means we can send
            // a SMS message to verify the phone number
            kickOffSMS();
        }

        // Check the current step's validity
        if (validity) {
            // If valid, enable the next button
            setDisableNextButton(false);

            // Check if we are on the last step
            if (currentStep >= steps.length - 1) {
                // Disable the next button and set the current step to the last step
                // since we should not go any further
                setDisableNextButton(true);
                setCurrentStep(steps.length - 1);
            } else {
                // Otherwise, go to the next step
                setCurrentStep(currentStep + 1);
            }
        } else {
            // If currently invalid, disable the next button
            setDisableNextButton(true);
        }

        // Enable the previous button since we are moving forward
        setDisablePreviousButton(false);
    }

    /**
     * Handle the submit button click event
     */
    function onSubmit(): void {
        // Get the unparsed model and parse it for submission
        const unparsedModel: StepperModel = {
            ...model,
            email,
            firstName,
            lastName,
            phone,
            zipCode,
        };
        let parsedModel: ParsedStepperModel = parseModel(unparsedModel);

        if (props['stepper-type'] === 'short') {
            parsedModel.isMailer = true;
        }

        if (parsedModel.isMailer && props['user-info'] !== undefined) {
            const {
                referenceId, middleInitial, suffix, address, city,
                stateAbbreviation, crrt, barcode, county, estimatedDebt,
                settled, newPayment, tier, segment, drop, tollFreeNumber,
                spanishTollFreeNumber, noticeDate, url, stateFull,
            } = props['user-info'];

            parsedModel = {
                ...parsedModel,
                referenceId, middleInitial, suffix, address, city,
                stateAbbreviation, crrt, barcode, county, estimatedDebt,
                settled, newPayment, tier, segment, drop, tollFreeNumber,
                spanishTollFreeNumber, noticeDate, url, stateFull,
            };
        }

        // Run submission
        submission(parsedModel);
    }

    return (
        <section className={styles['debt-advisors-stepper']}>
            <small>{`${currentStep + 1} of ${steps.length}`}</small>
            <div className={styles['steps-wrapper']}>
                {steps.map((step: Step) => {
                    const stepIndex: number = steps.findIndex((thisStep: Step) => thisStep.question === step.question);
                    const showStep: boolean = stepIndex === currentStep;

                    return (
                        <div key={step.question} className={`${styles['step']} ${showStep ? '' : styles['hidden']}`}>
                            <strong className={styles['question']}>{step.question}</strong>
                            {
                                step.stepType === 'multiple-choice' &&
                                    <MultipleChoiceField
                                        choices={step.choices !== undefined ? step.choices : []}
                                        field={step.property}
                                        setModel={setModel}
                                        setSteps={setSteps}
                                        model={model}
                                        nextStep={nextStep}
                                        steps={steps}
                                        currentStep={currentStep}
                                        stepperType={props['stepper-type']}
                                    />
                            }
                            {
                                step.stepType === 'currency' &&
                                    <Currency
                                        field={step.property}
                                        setModel={setModel}
                                        model={model}
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                    />
                            }
                            {
                                step.stepType === 'location' &&
                                    <Location
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                        zipCode={zipCode}
                                        setZipCode={setZipCode}
                                        setUSState={setUSState}
                                    />
                            }
                            {
                                step.stepType === 'name' &&
                                    <Name
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                        firstName={firstName}
                                        lastName={lastName}
                                        setFirstName={setFirstName}
                                        setLastName={setLastName}
                                    />
                            }
                            {
                                step.stepType === 'email' &&
                                    <Email
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                        email={email}
                                        setEmail={setEmail}
                                    />
                            }
                            {
                                step.stepType === 'phone' &&
                                    <Phone
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                        phone={phone}
                                        setPhone={setPhone}
                                    />
                            }
                            {
                                step.stepType === 'verify' &&
                                    <Verify
                                        steps={steps}
                                        currentStep={currentStep}
                                        setDisableNextButton={setDisableNextButton}
                                        code={code}
                                        setCode={setCode}
                                        disableVerifyField={disableVerifyField}
                                    />
                            }
                        </div>
                    );
                })}
            </div>
            {loading && <FontAwesomeIcon className="h-7 text-daa-purple animate-spin" icon={faCircleNotch} spin/>}
            {error !== '' && <small className={styles['error']}>{error}</small>}
            <div className={styles['stepper-controls-wrapper']}>
                <button
                    className={styles['previous-button']}
                    type="button"
                    onClick={() => previousStep()}
                    disabled={disablePreviousButton}
                ><FontAwesomeIcon className={styles['caret']} icon={faAngleLeft}/></button>
                {
                    steps.length - 1 !== currentStep &&
                    <button
                        className={styles['next-button']}
                        type="button"
                        onClick={() => nextStep()}
                        disabled={disableNextButton}
                    >{'Next'}<FontAwesomeIcon className={styles['caret']} icon={faAngleRight}/></button>
                }
                {
                    steps.length - 1 === currentStep &&
                    <button
                        className={styles['next-button']}
                        type="button"
                        onClick={() => onSubmit()}
                        disabled={disableNextButton}
                    >{`Call or Text Me Now`}</button>
                }
            </div>
        </section>
    );
}
