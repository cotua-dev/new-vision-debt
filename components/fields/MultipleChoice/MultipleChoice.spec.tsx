import { render, RenderResult } from '@testing-library/react';
import { StepperModel } from '../../Stepper/Stepper.interfaces';
import { initialStepperModel } from '../../Stepper/Stepper.initial';
import { MultipleChoiceField } from './index';
import { MultipleChoiceFieldProps } from './MultipleChoice.interfaces';

describe('MultipleChoiceField Component', (): void => {
    let renderResult: RenderResult;
    const props: MultipleChoiceFieldProps = {
        field: ['debtType'],
        choices: [],
        nextStep: () => {},
        setModel: () => {},
        model: initialStepperModel,
        currentStep: 0,
        steps: [],
    };

    beforeAll((): RenderResult => renderResult = render(<MultipleChoiceField {...props}/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
