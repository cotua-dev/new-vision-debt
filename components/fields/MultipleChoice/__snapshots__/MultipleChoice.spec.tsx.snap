// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleChoiceField Component renders 1`] = `
<div
  class="multiple-choice-wrapper"
/>
`;
